lib/explorer/polars_backend/lazy_frame.ex

```diff
@@ -7,16 +7,30 @@ defmodule Explorer.PolarsBackend.LazyFrame do
   alias Explorer.PolarsBackend.Native
   alias Explorer.PolarsBackend.Shared
   alias Explorer.PolarsBackend.DataFrame, as: Eager
-  alias Explorer.PolarsBackend.LazyFrame, as: PolarsLazyFrame
+  # alias Explorer.PolarsBackend.LazyFrame, as: PolarsLazyFrame
+  alias Explorer.PolarsBackend.LazyStack
 
   alias FSS.Local
   alias FSS.S3
 
   import Explorer.PolarsBackend.Expression, only: [to_expr: 1, alias_expr: 2]
 
-  defstruct resource: nil
+  defstruct polars_lazy_frame: nil, stack: []
 
-  @type t :: %__MODULE__{resource: reference()}
+  defmodule PolarsLazyFrame do
+    @moduledoc false
+
+    # This struct represents the lazy frame from the initialization
```

Comment:
Quick question: do we need to create a struct? Couldn't we just pass the reference around? Or does Rustler works better if we are passing structs?

Reply:
I think we can implement the a custom type with an encoder/decoder, or use one of the other `Nif*` macros. I think I will keep it this way for the PR and try to solve this in another opportunity. WDYT?

Reply:
We can do it later, yeah.